src/jit/hwintrinsicxarch.cpp

```diff
@@ -218,6 +228,39 @@ int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: get the number of arguments based on baseType
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//    baseType  -- var_types type of intrinsic
+//
+// Return Value:
+//     number of arguments
+//
+int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, var_types baseType)
+{
+    assert(intrinsic != NI_Illegal);
+    assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
+
+    int numArgs = hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
+    if (numArgs == -1)
```

Comment:
I suggest to have a flag `HW_Flag_UnfixedNumArgs` to make this behavior more reliable (we have had `HW_Flag_UnfixedSimdSize`).